src/Microsoft.TestPlatform.Common/Logging/InternalTestLoggerEvents.cs

```diff
@@ -167,7 +167,7 @@ internal void RaiseTestRunMessage(TestRunMessageEventArgs args!!)
         CheckDisposed();
 
         // Sending 0 size as this event is not expected to contain any data.
-        SafeInvokeAsync(() => TestRunMessage, args, 0, "InternalTestLoggerEvents.SendTestRunMessage");
+        SafeInvokeAsync(TestRunMessage, args, 0, "InternalTestLoggerEvents.SendTestRunMessage");
```

Comment:
Instead of creating action from each handler, we pass the handler directly. Then check if there are any subscribers, and if not, we don't put it on queue.